weblate/trans/models/unit.py

```diff
@@ -899,11 +902,22 @@ def run_checks(self, same_state=True, same_content=True, is_new=False):
             self.source_checks().values_list('check', flat=True)
         )
 
+        check_pair = set(FLAGS_PAIR_RE.findall('\n'.join(self.all_flags)))
+
         # Run all checks
         for check in checks_to_run:
             check_obj = CHECKS[check]
+
+            if len(check_pair) > 0 and hasattr(check_obj, 'check_value'):
+                check_value = max(
+                    {(x) for x in check_pair if x[0] == check},
+                    key=lambda v: v[1]
+                )[1]
+                setattr(check_obj, 'check_value', check_value)
```

Comment:
Passing the parameters as attributes is IMHO bad idea (the check objects are global, so they are shared between threads). I'd rather do the parsing in the check itself....